### PR TITLE
Feature/rpc and serialize fixes

### PR DESCRIPTION
**Pull request by volekerb**
### Problem
Missing unmarshal function for TXType and missing parametrization for "getblock" so it can return pure bytes
...

### Solution
this patchset
...

### CODE REVIEW

File: pkg/core/transaction/type.go [hunk 1]
```diff
@@ -52,3 +58,44 @@ func (t TXType) String() string {
 func (t TXType) MarshalJSON() ([]byte, error) {
 	return []byte(`"` + t.String() + `"`), nil
 }
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (t *TXType) UnmarshalJSON(data []byte) (err error) {
```
Comment (roman-khimov) [inline pkg/core/transaction/type.go]: `(error)`

File: pkg/core/transaction/type.go [hunk 2]
```diff
@@ -52,3 +58,44 @@ func (t TXType) String() string {
...
+	l := len(data)
+	if l < 2 || data[0] != '"' || data[l-1] != '"' {
+		return errors.New("wrong format")
+	}
+
+	*t, err = TXTypeFromString(string(data[1 : l-1]))
+	return
```
Comment (roman-khimov) [inline pkg/core/transaction/type.go]: `return err`

Comment (volekerb) [reply pkg/core/transaction/type.go]: fixed

File: pkg/core/transaction/type.go (hunk 1 above)
Comment (volekerb) [reply pkg/core/transaction/type.go]: fixed

**Comment by volekerb** [comment]
> And it would be nice to improve [4798d6c](https://github.com/nspcc-dev/neo-go/commit/4798d6c823319750b051289c12e06a9fe5874225) commit message, because it's adding unmarshalling for specific data type. Not critical, though.

done

**Pull request merged**